From 2917e9b6044a16b540fb605e0dc6a89d11649b93 Mon Sep 17 00:00:00 2001
From: Bharath <linux.bhar@gmail.com>
Date: Mon, 8 Jul 2019 06:06:18 -0600
Subject: [PATCH 2/4] mm: Add counters in vmstat for debugging

This patch adds vmstat counters which count the number of times kcompactd is woken
up and also counts the number of times the predictive algorithm detects a potential
external fragmentation event.

Signed-off-by: Bharath <linux.bhar@gmail.com>
---
 include/linux/vm_event_item.h | 2 ++
 mm/vmstat.c                   | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/include/linux/vm_event_item.h b/include/linux/vm_event_item.h
index 47a3441..afcd087 100644
--- a/include/linux/vm_event_item.h
+++ b/include/linux/vm_event_item.h
@@ -110,6 +110,8 @@ enum vm_event_item { PGPGIN, PGPGOUT, PSWPIN, PSWPOUT,
 		SWAP_RA,
 		SWAP_RA_HIT,
 #endif
+		KCOMPACTD_BY_PRED_MEM,
+		POSITIVE_RESULT,
 		NR_VM_EVENT_ITEMS
 };
 
diff --git a/mm/vmstat.c b/mm/vmstat.c
index a7d4933..b2664ea 100644
--- a/mm/vmstat.c
+++ b/mm/vmstat.c
@@ -1288,6 +1288,8 @@ int fragmentation_index(struct zone *zone, unsigned int order)
 	"swap_ra",
 	"swap_ra_hit",
 #endif
+	"kcompactd_by_pred_mem",
+	"positive_result",
 #endif /* CONFIG_VM_EVENTS_COUNTERS */
 };
 #endif /* CONFIG_PROC_FS || CONFIG_SYSFS || CONFIG_NUMA */
-- 
1.8.3.1

